### ISSUE TITLE
Need to declare color-scheme.

### ISSUE THREAD

**Issue by lukewarlow** [npmx-dev/npmx.dev#677]
Because npmx.dev doesn't currently use the color-scheme property on html (or the color-scheme meta tag) to advertise its support for color schemes, it fails to play nicely with forced dark mode.

To see examples of the issues load the website in chromium, open dev tools go to the rendering tab and toggle "Enable automatic dark mode" to on.

If your system is set to dark mode or your setting is set to dark mode explicitly, you'll notice that the logo is wrong, and switches lose contrast.

Similarly if your setting is explicitly set to light mode, you'll see the browser forces it to dark mode and again breaks various aspects of the UI.

In system mode the color-scheme property should be set to "light dark", in light mode it should be "light only", and in dark mode it should be "dark only".

**Comment by thasmo** [comment]
Looked it up because I haven't used the color-scheme property extensively.

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/meta/name/color-scheme

It states:
> For example, to indicate that a document prefers dark mode but also supports light mode:
`<meta name="color-scheme" content="dark light" />`

Furthermore, it states:
> `only light` indicates that the document only supports light mode, with a light background and dark foreground colors. `only dark` is invalid, because forcing a document to render in dark mode when it isn't compatible can result in unreadable content and all major browsers default to light mode if not otherwise configured.

So I guess we could use these:

**light and system mode**
`<meta name="color-scheme" content="light dark" />`

**dark mode**
`<meta name="color-scheme" content="dark light" />`

But I may also be wrong. 😅

**Comment by lukewarlow** [comment]
So it's not necessarily well documented what it means to "support" both modes. But when the theme selector is set to light mode or dark mode you don't support both modes, because you specifically only apply styles for one or the other.

The only keyword is the bit that stops browsers (or other tools) trying to override your rendering. If a user has specifically chosen light mode for your site that should win over the browser global force dark mode, hence 'light only'.

There's not actually a browser that offers a forced light mode but for consistency the same logic applies the other way around, hence 'dark only'.

Then in system mode where you do genuinely honour the user preference setting it to light dark tells the browser you'll take care of both, so again it doesn't try to do any force darkening.

**Comment by thasmo** [comment]
@lukewarlow, thanks for the explanation. Makes sense! 🙂

**Comment by thasmo** [comment]
@lukewarlow, that should do it. #689 🤞
